source/material_model/melt_boukare.cc

```diff
@@ -0,0 +1,1252 @@
+            }
+
+          properties.gibbs_energies[i] = G_Pref_Tf + intVdP;
+          properties.entropies[i] =  reference_entropies[i] + endmember_entropy_thermal_addition(in.temperature[q], i) + dintVdpdT;
+          properties.volumes[i] = reference_volumes[i] * (1 - a * (1. - std::pow(1. + b * (pressure - Pth), -c)));
+
+          properties.bulk_moduli[i] = reference_bulk_moduli[i] * (1. + b * (pressure - Pth))
+                                      * (a + (1. - a) * std::pow(1. + b * (pressure - Pth), c));
+
+          const double C_V0 = endmember_molar_heat_capacity(reference_temperature, i);
+          const double C_V = endmember_molar_heat_capacity(in.temperature[q], i);
+          properties.thermal_expansivities[i] = reference_thermal_expansivities[i] * (C_V / C_V0) *
+                                                1. / ((1. + b * (pressure - Pth)) *
+                                                      (a + (1. - a) * std::pow(1 + b * (pressure - Pth), c)));
+
+          const double Cp_ref = reference_specific_heats[i] + specific_heat_linear_coefficients[i] * in.temperature[q]
+                                + specific_heat_second_coefficients[i] * std::pow(in.temperature[q], -2.)
+                                + specific_heat_third_coefficients[i] * std::pow(in.temperature[q], -0.5);
+
+          const long double dSdT0 = reference_volumes[i] * reference_bulk_moduli[i] * std::pow(heat_capacity_ratio * reference_thermal_expansivities[i], 2.0)
+                                    * (std::pow(1. + b * (pressure - Pth), -1.-c) - std::pow(1. + b * (reference_pressure - Pth), -1.- c));
+
+          const double relative_T = Einstein_temperatures[i] / in.temperature[q];
+          const double dSdT = dSdT0 + dintVdpdT * (1 - 2./relative_T + 2./(std::exp(relative_T) - 1.)) * relative_T/in.temperature[q];
+
+          properties.heat_capacities[i] = Cp_ref + in.temperature[q] * dSdT;
+        }
+    }
+
+
+    template <int dim>
+    double
+    MeltBoukare<dim>::
+    endmember_thermal_energy (const double temperature,
+                              const unsigned int i) const
+    {
+      AssertThrow(temperature > 0.0,
+                  ExcMessage("The temperature has to be larger than 0, but it is "
+                             + std::to_string(temperature) + " for endmember " + std::to_string(i) + "."));
+
+      const double energy = 3. * number_of_atoms[i] * constants::gas_constant * Einstein_temperatures[i]
+                            * (0.5 + 1. / (std::exp(Einstein_temperatures[i] / temperature) - 1.0));
+
+      return energy;
+    }
+
+
+    template <int dim>
+    double
+    MeltBoukare<dim>::
+    endmember_molar_heat_capacity (const double temperature,
+                                   const unsigned int i) const
+    {
+      AssertThrow(temperature > 0.0,
+                  ExcMessage("The temperature has to be larger than 0!"));
+
+      const double relative_T = Einstein_temperatures[i] / temperature;
```

Comment:
could you add the same replacement to the function above? only to make them easier to compare.